### ISSUE TITLE
(3.0 Alpha 2) Heading sizes are overriden in Gutenberg

### ISSUE THREAD

**Issue by wplit** [soflyy/oxygen-bugs-and-features#430]
**Describe the bug**
When headings have been styled using classes, the CSS is not correctly added when viewing in Gutenberg. Instead we always see the heading size and font weight as set in our global settings, not those individually styled with classes. 

**Screenshots**
Example from atomic design set where the class 'atomic-secondary-heading' is being overridden and we're seeing 60px heading (60px coming from global style for h1 tags), instead of the 48px set in Oxygen via the 'atomic-secondary-heading' class. Also, we're getting a font weight of 500 instead of 400.

<img width="1599" alt="Screen Shot 2019-07-21 at 8 06 44 am" src="https://user-images.githubusercontent.com/43051571/61584667-13100200-ab8f-11e9-9de7-4f9525b9761f.png">

I see how all styles can't be exactly matched. But font sizes can really change a layout quite a bit, especially when we have some really small h1's but are being shown as 60px.

**Comment by louisreingold** [comment]
Maybe we should prefix all the classes with .editor-styles-wrapper?

Working around the quirks of Gutenberg has been a nightmare. I hate to add more "hacks" to Oxygen's code to accommodate Gutenberg's inexplicable decision not to use an iframe, although at this point we've already added a ton of them so I guess what's one more?

**Comment by wplit** [comment]
I can imagine. 

I don't think there would be a way around adding the .editor-styles-wrapper class. That's the way I've seen theme's do it where they've added support for editor styles. 

Any classes in their 'editor-style.css' stylesheet are automatically prefixed with the .editor-styles-wrapper by WordPress.

Source: Genesis child theme, adding rule '.bigger-font { font-size: 120px; } ' in the editor.style.css file in the theme, actually outputs as `.editor-styles-wrapper .bigger-font { font-size: 120px; }` while in editor.

<img width="1000" alt="Screen Shot 2019-07-21 at 12 19 58 pm" src="https://user-images.githubusercontent.com/43051571/61586224-27b1c180-abb2-11e9-949c-5281626a7fc6.png">

**Comment by louisreingold** [comment]
@wplit how's this?

[oxygen-3p0_18b4c6f.zip](https://github.com/soflyy/oxygen-bugs-and-features/files/3424313/oxygen-3p0_18b4c6f.zip)
[oxygen-gutenberg-5f1df6e.zip](https://github.com/soflyy/oxygen-bugs-and-features/files/3424314/oxygen-gutenberg-5f1df6e.zip)

**Comment by wplit** [comment]
Yes, this seems to be working as expected from what I've tried so far.

**Comment by wplit** [comment]
Ok now, it's not working. When i went back into the page, none of the styles were being added at all. I'll try to recreate.

**Comment by wplit** [comment]
Error on front end.. the `body:not(.wp-admin)` from Oxygen.css is more specific than the global body font settings in universal, so the global body font size no longer works.

[Demo site with body font size set as 40px, but showing as 16px](http://glamorous-dolphin.w5.wpsandbox.pro/)

<img width="517" alt="Screen Shot 2019-07-24 at 10 12 08 am" src="https://user-images.githubusercontent.com/43051571/61755578-a3527f00-adfb-11e9-9390-dd50bad4d497.png">

**Comment by wplit** [comment]
I've found out the situation when the editor styles are failing as I mentioned in one of my previous comments.. It only happens when working locally (with Local by Flywheel) with SSL installed (this is why I couldn't reproduce in sandbox site). 

In this case, none of the classes styling or global styles get added to the editor at all, just from the IDs. So the layout is completely out of whack. 

I tested with live sites, both SSL and none, it works fine. So does locally without SSL. But locally with SSL breaks everything.

<img width="1400" alt="Screen Shot 2019-07-24 at 11 30 38 am" src="https://user-images.githubusercontent.com/43051571/61758057-8f604a80-ae06-11e9-8922-58f3d803416f.png">

**Comment by louisreingold** [comment]
OK, so other than the SSL, the remaining issue is the :not(.wp-admin)? If you edit that selector to just begin with body instead of body:not(.wp-admin) do things work as expected, including in the WP admin?

**Comment by wplit** [comment]
Yes, that fixes it. Just having is as 'body' in Oxygen.css means it becomes `.editor-styles-wrapper` in the editor, which is then correctly overridden by the body font size settings, which also outputs `.editor-styles-wrapper` but comes later. So it works.
